### ISSUE TITLE
[suggestion] Seperate fleet personalities for...

### ISSUE THREAD

**Issue by toilethinges** [endless-sky/endless-sky#1147]
...fighters/drones and independent ships? Similarly to the "fighters"/"names" attributes in fleets.

This way, we could have swarming fighters but not swarming interceptors/light warships...

Edit: what does it mean by, in the description of swarming, "ships of this type"? That suggests that you can already do what I suggested above..

**Comment by tehhowch** [comment]
One way to do this would be to modify the `fighters` node.
Currently, it is used solely to indicate the naming phrase for a fleet's carried ships. Separate behaviors can be achieved in missions by use of a separate NPC block, but for randomly placed NPCs, this cannot be done (as there is no way to tell one NPC block to use the same system chosen via LocationFilter for a different NPC block) except for origin and destination NPCs.

Current:
```
npc
    government Republic
    personality heroic
    fleet
        names "republic capital"
        variant
            Cruiser
npc
    government Republic
    personality timid
    fleet
        fighters "republic fighter"
        variant
            "Combat Drone" 4
```
Proposed
```
npc
    government Republic
    personality heroic
    fleet
        names "republic capital"
        fighters <name phrase is still supported>
            names <can also be here>
            personality timid
        variant
            Cruiser
            "Combat Drone" 4
```

Overriding the `fighters` token is one method. Does anyone have alternate methods of achieving this goal? Another method is to always make fighters `timid` (or perhaps each government has a specific carried ship personality, to allow some differentiation between organizations), but that would remove a creator's ability to tune ship behavior (as suggested in OP here e.g, swarming)

**Comment by tehhowch** [comment]
Using just the above approach won't actually help with mission NPCs, but would allow for the randomly encountered fleets to have "sensible" fighters. This is because NPC::Instantiate applies its personality to all ships, whether or not one was defined in the `fleet` node inside the NPC block (and the NPC loading code has no idea that a personality node is even defined inside the fleet node). This also means that even if you don't set a personality in a mission's NPC block, any personality defined in the fleet is still overridden (because the NPC has a default personality).

What could maybe work would be to 1) allow the fleet class to export its personality, 2) add a keyword for NPC::Load that indicates it should read in the personality from the fleet class (`fleet personality`?), and 3) make the previous post's modifications to the `fighters` node.

A forward-looking improvement would be to convert the references to personalities in NPC and Fleet into mappings between a ship class and a personality. This would open up setting different personalities for drones, for fighters, for light warships, for heavy warships, etc. The difficulty then is in how to effectively and easily read and write these specifics from saves and mission files - feedback welcomed.

**Comment by warp-core** [comment]
> A forward-looking improvement would be to convert the references to personalities in NPC and Fleet into mappings between a ship class and a personality. This would open up setting different personalities for drones, for fighters, for light warships, for heavy warships, etc. The difficulty then is in how to effectively and easily read and write these specifics from saves and mission files - feedback welcomed.

Give `ship` a child `personality` node?

**Comment by Zitchas** [comment]
Only if we can override the ship's personalities with the typical NPC fleet definitions. I don't think all of any ship should be restricted to a single behavior.

**Comment by warp-core** [comment]
I only meant for instantiated ships, so ones being written to the savefile. It wouldn't be a part of the ship definition.

**Comment by warp-core** [comment]
Who merged #9029 and didn't close this?
